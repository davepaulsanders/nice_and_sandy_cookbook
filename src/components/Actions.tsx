interface ActionsProps {
	label: string;
	action: "submit" | "delete";
	type: "submit" | "reset" | "button";
	onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
	}
const Actions = ({label, action, type, onClick}: ActionsProps) => {
	return (
		<button className={`text-xl ${action === "submit" ? "bg-medg hover:bg-lgg" : "bg-deleteColor hover:bg-deleteColorHover"} \
			hover:text-slate-200 py-3 px-10 rounded-md`} type={type} onClick={onClick}>{label}</button>
		)
	}

export default Actions;
