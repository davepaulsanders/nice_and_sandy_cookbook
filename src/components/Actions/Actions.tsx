interface ActionsProps {
	label: string;
	type: "link" | "button";
	button?: {type: string, action: string}
	href: string;
	onClick?: (e: React.MouseEvent<HTMLButtonElement> | React.MouseEvent<HTMLAnchorElement>) => void;
	}
const Actions = ({label, type, button, href, onClick}: ActionsProps) => {
	if (type === "button" && button) {
	return (
		<button className={`hover:cursor-pointer text-xl ${button.action === "submit" ? "bg-medg hover:bg-lgg" : "bg-deleteColor hover:bg-deleteColorHover"} \
			hover:text-slate-200 py-3 px-10 rounded-md`} type={type} onClick={onClick}>{label}</button>
		)
		} else {
			return (
				<a target="_blank" href={href} rel="noopener noreferrer" className="text-xl hover:underline hover:cursor-pointer">{label}</a>
			)
		}
		
	}

export default Actions;
