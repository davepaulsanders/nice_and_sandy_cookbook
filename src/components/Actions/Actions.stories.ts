import type { Meta, StoryObj } from '@storybook/react';
import { fn } from '@storybook/test'; 
 
import Actions from './Actions';
 
const meta: Meta<typeof Actions> = {
  component: Actions,
};
 
export default meta;
type Story = StoryObj<typeof Actions>;
export const SubmitAction: Story = {
  args: {
    label: 'Add recipe',
	button: {type: "submit", action: "submit"},
	type: "button",
	onClick: fn()
  },
};
export const DeleteAction: Story = {
  args: {
    label: 'Remove recipe',
	button: {type: "delete", action: "delete"},
	type: "button",
	onClick: fn()
}
};
export const Link: Story = {
  args: {
    label: 'TikTok Pasta',
	type: "link",
	onClick: fn()
}
};
