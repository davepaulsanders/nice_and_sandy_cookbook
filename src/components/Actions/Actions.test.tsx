import Actions from "./Actions" 
import { expect, test } from 'vitest'
import {render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/dom'
import { vi } from 'vitest'
const mockFunc = vi.fn()
test('<Actions /> as link renders', () => {
	const { container } = render(<Actions label="test" type="link" href="www.google.com"/>)
	expect(container).toMatchSnapshot()
})
test('<Actions /> as submit button renders', () => {
	const { container } = render(
        <Actions 
            label="test" 
            button={{ type: "submit", action: "submit" }} 
            type="button" 
            onClick={vi.fn()} 
        />
    );
	expect(container).toMatchSnapshot()
})
test('<Actions /> as delete button renders', () => {
	const { container } = render(
        <Actions 
            label="test" 
            button={{ type: "button", action: "delete" }} 
            type="button" 
            onClick={vi.fn()} 
        />
    );
	expect(container).toMatchSnapshot()
})
test('<Actions /> sends onClick', async () => {
	const { container } = render(
        <Actions 
            label="test" 
            button={{ type: "button", action: "delete" }} 
            type="button" 
            onClick={mockFunc} 
        />
    );
	await userEvent.click(screen.getByText("test"))
	expect(mockFunc).toHaveBeenCalledTimes(1)
})
