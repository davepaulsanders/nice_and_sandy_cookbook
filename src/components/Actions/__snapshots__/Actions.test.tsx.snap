// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Actions /> as delete button renders 1`] = `
<div>
  <button
    class="hover:cursor-pointer text-xl bg-deleteColor hover:bg-deleteColorHover 			hover:text-slate-200 py-3 px-10 rounded-md"
    type="button"
  >
    test
  </button>
</div>
`;

exports[`<Actions /> as link renders 1`] = `
<div>
  <a
    class="text-xl hover:underline hover:cursor-pointer"
    href="www.google.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    test
  </a>
</div>
`;

exports[`<Actions /> as submit button renders 1`] = `
<div>
  <button
    class="hover:cursor-pointer text-xl bg-medg hover:bg-lgg 			hover:text-slate-200 py-3 px-10 rounded-md"
    type="button"
  >
    test
  </button>
</div>
`;

exports[`<Actions /> as submit button renders 2`] = `
<div>
  <button
    class="hover:cursor-pointer text-xl bg-deleteColor hover:bg-deleteColorHover 			hover:text-slate-200 py-3 px-10 rounded-md"
    type="button"
  >
    test
  </button>
</div>
`;
